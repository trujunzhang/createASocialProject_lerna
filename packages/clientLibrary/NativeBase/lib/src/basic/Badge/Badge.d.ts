import { ReactNative, BsStyle } from '@appUtils/rnCommonProps';
export interface BadgeProps extends ReactNative.ViewProps, BsStyle {
    textStyle?: any;
}
