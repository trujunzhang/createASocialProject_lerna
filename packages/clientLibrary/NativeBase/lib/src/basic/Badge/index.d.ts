declare const StyledBadge: any;
export { StyledBadge as Badge };
