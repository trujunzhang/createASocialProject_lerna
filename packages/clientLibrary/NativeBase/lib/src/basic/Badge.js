import React, { Component } from "react";
import PropTypes from "prop-types";
import { View, ViewPropTypes } from "react-native";

import { connectStyle } from "@app/native-base-shoutem-theme";
import { platformVariables as variable } from '@app/native-base-variables';
import mapPropsToStyleNames from "../utils/mapPropsToStyleNames";

class Badge extends Component {
  render() {
    return (
      <View ref={c => (this._root = c)} {...this.props}>
        {this.props.children}
      </View>
    );
  }
}

Badge.propTypes = {
  ...ViewPropTypes,
  style: PropTypes.oneOfType([
    PropTypes.object,
    PropTypes.number,
    PropTypes.array
  ])
};

const StyledBadge = connectStyle("NativeBase.Badge", {}, mapPropsToStyleNames)(
  Badge
);

export { StyledBadge as Badge };
