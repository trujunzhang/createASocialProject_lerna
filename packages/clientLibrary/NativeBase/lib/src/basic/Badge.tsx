import * as React from 'react'
import PropTypes from "prop-types";
import { View, ViewPropTypes } from "react-primitives";

import { connectStyle } from "@app/native-base-shoutem-theme";
import { platformVariables as variable } from '@app/native-base-variables';
import mapPropsToStyleNames from "../utils/mapPropsToStyleNames";

class Badge extends React.Component {
  private _root: any
  render() {
    return (
      <View ref={c => (this._root = c)} {...this.props}>
        {this.props.children}
      </View>
    );
  }
}

// Badge.propTypes = {
//   ...ViewPropTypes,
//   style: PropTypes.oneOfType([
//     PropTypes.object,
//     PropTypes.number,
//     PropTypes.array
//   ])
// };

const StyledBadge = connectStyle("NativeBase.Badge", {}, mapPropsToStyleNames)(
  Badge
);

export { StyledBadge as Badge };
