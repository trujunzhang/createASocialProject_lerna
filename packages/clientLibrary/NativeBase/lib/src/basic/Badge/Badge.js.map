{"version":3,"file":"Badge.js","sourceRoot":"","sources":["Badge.ts"],"names":[],"mappings":""}