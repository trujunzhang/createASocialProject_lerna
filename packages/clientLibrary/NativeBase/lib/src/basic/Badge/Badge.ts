import * as React from 'react'

import { ReactNative, BsStyle } from '@appUtils/rnCommonProps'

export interface BadgeProps extends ReactNative.ViewProps, BsStyle {
  // Others
  textStyle?: any
}
