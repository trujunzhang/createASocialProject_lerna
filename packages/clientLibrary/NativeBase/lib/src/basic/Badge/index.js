import * as React from 'react';
import { View } from 'react-primitives';
import { connectStyle } from 'native-base-shoutem-theme';
import { mapPropsToStyleNames } from '@app/tools';
class Badge extends React.Component {
    render() {
        return (React.createElement(View, Object.assign({ ref: (c) => (this._root = c) }, this.props), this.props.children));
    }
}
// Badge.propTypes = {
//   ...ViewPropTypes,
//   style: PropTypes.oneOfType([
//     PropTypes.object,
//     PropTypes.number,
//     PropTypes.array
//   ])
// };
const StyledBadge = connectStyle('NativeBase.Badge', {}, mapPropsToStyleNames)(Badge);
export { StyledBadge as Badge };
//# sourceMappingURL=index.js.map