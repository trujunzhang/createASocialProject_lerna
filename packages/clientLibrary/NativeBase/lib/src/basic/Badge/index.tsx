import * as React from 'react'
import * as PropTypes from 'prop-types'
import { View } from 'react-primitives'

import { connectStyle } from 'native-base-shoutem-theme'

import { IThemeVariables } from '@app/tools' // typings

import { platformVariables as variables } from '@appVariables/theme/variables/platform' // variables // [variables]
import { mapPropsToStyleNames } from '@app/tools'

import { BadgeProps } from './Badge'

class Badge extends React.Component<BadgeProps, any> {
  private _root: any
  render() {
    return (
      <View ref={(c) => (this._root = c)} {...this.props}>
        {this.props.children}
      </View>
    )
  }
}

// Badge.propTypes = {
//   ...ViewPropTypes,
//   style: PropTypes.oneOfType([
//     PropTypes.object,
//     PropTypes.number,
//     PropTypes.array
//   ])
// };

const StyledBadge = connectStyle('NativeBase.Badge', {}, mapPropsToStyleNames)(Badge)

export { StyledBadge as Badge }
