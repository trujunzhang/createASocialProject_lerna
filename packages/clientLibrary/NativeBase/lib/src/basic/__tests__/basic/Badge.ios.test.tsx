import 'react-primitives'
import * as React from 'react'
import * as renderer from 'react-test-renderer'
import { Badge } from '../../Badge'

// Note: test renderer must be required after react-native.

// jest.mock("Platform", () => {
//   const Platform = require.requireActual("Platform");
//   Platform.OS = "ios";
//   return Platform;
// });

it('renders regular badge', () => {
  const tree = renderer.create(<Badge>2</Badge>).toJSON()
  expect(tree).toMatchSnapshot()
})
