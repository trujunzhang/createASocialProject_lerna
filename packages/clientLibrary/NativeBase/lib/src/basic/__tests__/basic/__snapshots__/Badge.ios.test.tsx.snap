// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders regular badge 1`] = `
<view
  name="View"
  style={Object {}}
>
  2
</view>
`;
